File: server/src/main/java/com/bomberman/entities/mapobject/pickup/Pickup.java
package com.bomberman.entities.mapobject.pickup;

import com.bomberman.constants.Configuration;
import com.bomberman.entities.Position;
import com.bomberman.entities.mapobject.MapObject;
import com.bomberman.entities.mapobject.movable.Player;

public abstract class Pickup extends MapObject {

  public Pickup() {}

  public Pickup(final Position position) {
    super(position);
  }

  public abstract void apply(Player player);

  @Override
  public int getTextureWidth() {
    return Configuration.TEXTURE_SIZE;
  }

  @Override
  public int getTextureHeight() {
    return Configuration.TEXTURE_SIZE;
  }
}

File: server/src/main/kotlin/com/bomberman/entity/mapobject/pickup/Pickup.kt
package com.bomberman.entity.mapobject.pickup

import com.bomberman.TEXTURE_SIZE
import com.bomberman.entity.Position
import com.bomberman.entity.mapobject.MapObject
import com.bomberman.entity.mapobject.movable.Player

abstract class Pickup(position: Position) : MapObject(position) {

    abstract fun apply(player: Player)

    abstract val className: String

    override val textureHeight: Int
        get() = TEXTURE_SIZE

    override val textureWidth: Int
        get() = TEXTURE_SIZE
}